### PR TITLE
[SELC-4738] feat: add flag to force institution creation on every onboarding

### PR DESCRIPTION
**Pull request by KevinSi96**
#### List of Changes

added flag that forces the creation of the institution on every onboarding

#### Motivation and Context

Solve's bug raised from the implementation done to onboard pnpg institutions

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/onboarding-functions/src/main/resources/application.properties [hunk 1]
```diff
@@ -16,7 +16,7 @@ quarkus.index-dependency.onboarding.artifact-id=onboarding-sdk-common
 onboarding-functions.retry.max-attempts = ${RETRY_MAX_ATTEMPTS:5}
 onboarding-functions.retry.first-retry-interval = ${FIRST_RETRY_INTERVAL:5}
 onboarding-functions.retry.backoff-coefficient = ${BACKOFF_COEFFICIENT:5}
-
+onboarding-function.force-institution-persist= ${FORCE_INSTITUTION_PERSIST:false}
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/resources/application.properties:19]: you should added `FORCE_INSTITUTION_PERSIST=true` under infra/container_apps for pnpg environment

File: apps/onboarding-functions/src/main/resources/application.properties [hunk 2]
```diff
@@ -16,7 +16,7 @@ quarkus.index-dependency.onboarding.artifact-id=onboarding-sdk-common
 onboarding-functions.retry.max-attempts = ${RETRY_MAX_ATTEMPTS:5}
 onboarding-functions.retry.first-retry-interval = ${FIRST_RETRY_INTERVAL:5}
 onboarding-functions.retry.backoff-coefficient = ${BACKOFF_COEFFICIENT:5}
-
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/resources/application.properties:19]: you could add some comments to explain why we need this variable

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 3]
```diff
@@ -89,24 +89,31 @@ public class CompletionServiceDefault implements CompletionService {
     @ConfigProperty(name = "onboarding-functions.persist-users.active")
     private boolean isUserMSActive;
 
+    @ConfigProperty(name = "onboarding-function.force-institution-persist")
+    private boolean forceInstitutionCreation;
+
     @Override
     public String createInstitutionAndPersistInstitutionId(Onboarding onboarding) {
+        InstitutionResponse institutionResponse = null;
+        if(forceInstitutionCreation){
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java:98]: you could add some comments to explain why we perform this if

**Review by manuraf — CHANGES_REQUESTED** [review]

File: apps/onboarding-functions/src/main/resources/application.properties (hunk 1 above)
Comment (KevinSi96) [reply apps/onboarding-functions/src/main/resources/application.properties:19]: should we set this to true straight away?

Comment (manuraf) [reply apps/onboarding-functions/src/main/resources/application.properties]: no, default false is okay!

**Review by manuraf — APPROVED** [review]

**Pull request merged**